gc/base/EnvironmentBase.cpp

```diff
@@ -566,3 +573,14 @@ MM_EnvironmentBase::restoreObjects(omrobjectptr_t *objectPtrIndirect)
 		Assert_MM_unreachable();
 	}
 }
+
+void
+MM_EnvironmentBase::checkColorAndMark(omrobjectptr_t objectPtr)
```

Comment:
not sure where/how this method is to be used, but consider moving this method to some other class (generally, Env class should be fairly light of decision making code specific to GC policies etc).

Reply:
This method is meant to to premark on per-object level, ideally the entire TLH should be pre-marked. However, in case we don't do a TLH allocation we must premark the object itself - similar to what Metronome does for large object allocations, when it can't allocate in a pre-marked cell.

Reply:
Moved to ParallelCollector